src/core/unif_rule.ml

```diff
@@ -71,3 +71,37 @@ let rec p_unpack : p_term -> (p_term * p_term) list = fun eqs ->
       else if id s = "#equiv" then [(v, w)] else
       assert false (* Ill-formed term. *)
   | _                               -> assert false (* Ill-formed term. *)
+
+let prod : sym =
+  Sign.add_symbol sign Public Defin (Pos.none "#prod") Kind []
+
+let rec prod_obj : term -> term = fun t ->
+  match unfold t with
+  | Appl(t,u) -> Appl(prod_obj t, prod_obj u)
+  | Prod(a,b) ->
+      let (x, b) = Bindlib.unbind b in
+      let b = lift (prod_obj b) in
+      let b = Bindlib.unbox (Bindlib.bind_var x b) in
+      Appl(Symb(prod), Abst(prod_obj a,b))
+  | Abst(a,b)         ->
```

Comment:
Since we need to transform top produtcs only (right?), the cases Appl and Abst are useless.